### DISCUSSION TITLE
Side effects of shadows

### DISCUSSION THREAD

**Discussion by rainergericke** [vsg-dev/VulkanSceneGraph#1081] [Q&A] [answered]
Looking in details of our vehicle simulation we have found some side effects, when shadows are enabled.
The following images show the same vehicle at different camera positions. The 'sun' (directional light) is shining from the left.
Image 1: No shadows - sunny side
![hmmwv_left_sunnyside_noshadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/0900f50e-b2fe-44d0-9d27-44512edd84e9)
Image 2: Shadows - sunny side
![hmmwv_left_sunnyside_shadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/22513c3d-c89d-49e7-bc2a-08ad3973cef0)
The sunny sides are correct. When you look at the rims, you see detail like the wheel screws. You can also see details on the car body.
Image 3: No shadows - dark side
![hmmwv_right_dark_side_noshadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/ff847ba0-4210-4b01-9c65-87a31eee6207)
Image 4: Shadows - dark side
![hmmwv_right_dark_side_shadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/f573e219-4913-46bb-81a7-b70c5f3a388e)
Without shadows we can see all details also on the dark side (image 3), while using shadows we get flat shaded surfaces on the dark side (image 4).
Image 5: No shadows - rear side
![hmmwv_rear_noshadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/2e8022b6-77c3-4034-83cf-ec17ebf71862)
Image 6: Shadows - rear side
![hmmwv_rear_shadow](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/af5e448d-6c96-4868-83e6-56131ebebd9e)
Looking at the rear side we get PBR shading at the suspension ('axle') parts (image 5) without shadows and flat shading with shadows enabled (image 6).

Is the a possibility to avoid this?

**Comment by robertosfield** [comment]
The shadowed scenes look more "correct" in terms of how light works on scenes like this, from the your description you want some of the cues you get from an unshadowed scene even iif those lighting cues are not what you'd see in the real world.

I suspect if the models had some texturing on rather that flat coloured you'd get more visual cues about the structure in the shadowed region.

However, in the real world even flat surfaces you'd get some cues through scatter and ambient occlusion,  Image based lighting would be something that could help with providing another source of light to provide some more lighting cues on the shadowed sides.  

Short of that an extra light source that doesn't have shadows, such as an dim overhead light might give you enough cues.

**Comment by rainergericke** [comment]
Sounds good, we will try the dimmed overhead light!

**Reply by robertosfield** [reply]
Another approach would be to provide a modified fragment shader that changes the shadow so that it doesn't 100% block a light associated with the shadow map, but just dims it to something like 80%, this will allow some of the light to bleed through. It's physically correct but might give you the effect you are after without the extra cost of the over head light.

Another approach would be to have a dimmed head light that doesn't have a shadow map associated with it.

**Comment by rainergericke** [comment, marked as answer]
Thanks, introducing the helper headlight make the vehicle look much better!
![hmmwv](https://github.com/vsg-dev/VulkanSceneGraph/assets/31243285/0064f942-ee32-4c66-ac78-3b349b0dce09)

**Comment by AnyOldName3** [comment]
I'd just like the weigh in that Ambient Occlusion should help a lot with this kind of thing without the costs of image-based lighting (like having a plausible image to use for lighting, which can be infeasible if you've got a dynamic environment and it's too varied to get away with a single static environment map). When the direct light is entirely shadowed, you're only seeing the ambient light, so problems with it, like it reaching places it shouldn't, will be particularly apparent.

For this particular scene, you could probably use a static environment map for image-based lighting, though, as apart from the texture tiling, it looks like nothing would change as the humvee moved.

**Comment by rainergericke** [comment]
We will keep this in mind for future development. Our software can simulate physical systems like vehicles and often we want to show technical aspects by means of false colors. This can already be done now and works well. Showing vehicles in a realistic environment is on our todo list yet. That's also the reason to use PBR from the beginning.

Thanks!
